src/currentWeather/SearchHistory.tsx

```diff
@@ -0,0 +1,40 @@
+import "./SearchHistory.css";
+
+type SearchHistoryProps = {
+  searchHistory: string[];
+  onSetLocation: React.Dispatch<React.SetStateAction<string>>;
+  onClick: (location: string) => Promise<void>;
```

Comment:
No, onClick shouldn't be a promise type, because search history doesn't give a fuck if someone uses this prop for async stuff